data/trie/interface.go

```diff
@@ -0,0 +1,53 @@
+package trie
+
+import (
+	"io"
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/hashing"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+)
+
+type node interface {
```

Comment:
I agree, there are a lot of methods to this interface, but I can't find another solution.The idea is that the trie holds a pointer to the root, which can be either branchNode, extensionNode or leafNode. Instead of using a lot of switch - case statements, I chose to use polymorphism, so each node type implements the node interface.